File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        
        Map<String,List<String>> mp=new HashMap<>();

        for(String str:strs){
            char charr[]=str.toCharArray();
            Arrays.sort(charr);
            String st=String.valueOf(charr);
            if(!mp.containsKey(st)){
                mp.put(st,new ArrayList<>());
               
            }
             mp.get(st).add(str);

        }

        List<List<String>> ans=new ArrayList<>(mp.values());

        return ans;
    }
}

File: Data Structures & Algorithms/string-matching-in-an-array/submission-0.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:

        words.sort()
        st=set()

        for i in range(len(words)):
            s=words[i]
            for val in words[1:]:
                if len(val)==len(s):
                    continue
                if s in val:
                    st.add(s)

        return list(st)
        